### DISCUSSION TITLE
Using custom colours for embedding

### DISCUSSION THREAD

**Discussion by AAA-3** [theislab/scvelo#654] [Q&A] [answered]
Hello!

I am imported a colour palette from my Seurat file and stored the colours in the  same order as the clusters:

![image](https://user-images.githubusercontent.com/68068335/130446855-ae445209-910e-42fd-8300-9caa1d0e98b4.png)

In trying to use those colours to label my clusters,:

`scv.pl.velocity_embedding_stream(WT, basis='tsne', color='Clustername', legend_loc='right margin', palette=WT.uns['ClusterName_colors'])`

I get this warning:
![image](https://user-images.githubusercontent.com/68068335/130447183-58704a16-7516-4789-a951-8ddd1cf29ad9.png)

How can I call  my seurat colours in scvelo? I checked #103 and #183 but it doesn't seem to work...

#### Versions:
scvelo==0.2.3  scanpy==1.8.1  anndata==0.7.6  loompy==3.0.6  numpy==1.21.2  scipy==1.6.3  matplotlib==3.4.3  sklearn==0.24.2  pandas==1.3.2  

Numba: Attempted to fork from a non-main thread, the TBB library may be in an invalid state in the child process.
ERROR: XMLRPC request failed [code: -32500]
RuntimeError: PyPI's XMLRPC API is currently disabled due to unmanageable load and will be deprecated in the near future. See https://status.python.org/ for more information.

**Comment by WeilerP** [comment, marked as answer]
@AAA-3, you don't set the argument `palette`. You simply need to specify the column name by which to color the scatter plot. The colors are inferred automatically.

**Note:** Running your cell changes the value of `adata.uns['ClusterName_colors']`. Consequently, to color the embedding with your original colors, you need to reload the data and remove the argument `palette`.

**Reply by AAA-3** [reply]
When I reloaded the data and ran the code:
`scv.pl.velocity_embedding_stream(WT, basis='tsne', color='Clustername', legend_loc='right margin')`
I got the same output. I doublechecked my colours as well and they were in the correct order (same as above) 

The colours I expect (not in the same order as the adata):
![image](https://user-images.githubusercontent.com/68068335/130456460-bcee2a8f-06cc-4738-93b4-654bb3443ec2.png)

**Reply by WeilerP** [reply]
@AAA-3, would you please change the field name in `adata.uns` from `'ClusterName_colors'` to `'Clustername_colors'` and check the result?

**Reply by AAA-3** [reply]
Worked! Thanks :D
